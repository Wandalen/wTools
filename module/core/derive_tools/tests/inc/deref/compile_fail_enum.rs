extern crate derive_tools_meta;
// # Test Matrix for `Deref` on Enums (Compile-Fail)
//
// This matrix documents test cases for ensuring the `Deref` derive macro correctly
// rejects enums, as `Deref` is only applicable to structs with a single field.
//
// | ID   | Item Type | Expected Error Message                                   |
// |------|-----------|----------------------------------------------------------|
// | CF1.1 | Enum      | "Deref cannot be derived for enums. It is only applicable to structs with a single field." |

#[ allow( dead_code ) ]
#[ derive( derive_tools_meta::Deref ) ]
enum MyEnum
{
  Variant1( bool ),
  Variant2( i32 ),
}

fn main() {}