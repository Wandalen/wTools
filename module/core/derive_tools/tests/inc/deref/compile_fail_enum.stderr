error: Deref cannot be derived for enums. It is only applicable to structs with a single field.
  --> tests/inc/deref/compile_fail_enum.rs:11:1
   |
11 | / #[ allow( dead_code ) ]
12 | | #[ derive( derive_tools_meta::Deref ) ]
13 | | enum MyEnum
...  |
16 | |   Variant2( i32 ),
17 | | }
   | |_^
